db_test.go

```diff
@@ -108,6 +108,30 @@ func TestDBConnectWithStartupNotice(t *testing.T) {
 	require.NoError(t, db.Ping(context.Background()), "must successfully ping database with long application name")
 }
 
+func TestBeforeConnect(t *testing.T) {
```

Comment:
I would add another test that does tests the before connect with multiple go routines to make sure that a race condition is not triggered by accident. A race condition in a before connection hook for an ORM could cripple an applicaiton.

Reply:
I think a good way to do this would be to set the options to have a max pool of 1. And then try to send multiple queries concurrently in a few go routines.

This should trigger a concurrent call of the BeforeConnect method; and given the `-race` flag should be a sufficient enough smoke test for most use cases.